internal/services/compute/managed_disk_resource.go

```diff
@@ -673,11 +685,15 @@ func resourceManagedDiskRead(d *pluginsdk.ResourceData, meta interface{}) error
 				d.Set("logical_sector_size", creationData.LogicalSectorSize)
 			}
 
-			imageReferenceID := ""
-			if creationData.ImageReference != nil && creationData.ImageReference.ID != nil {
-				imageReferenceID = *creationData.ImageReference.ID
+			galleryImageReferenceId := ""
+			imageReferenceId := ""
+			if galleryImageReference := creationData.GalleryImageReference; galleryImageReference != nil && galleryImageReference.ID != nil {
+				galleryImageReferenceId = *galleryImageReference.ID
+			} else if imageReference := creationData.ImageReference; imageReference != nil && imageReference.ID != nil {
```

Comment:
Use `else if` so that only one of the `image_reference_id` and `gallery_image_reference_id` is assigned.
When galleryImageReferenceId is used, the API assigns the value to imageReferenceId as well, which should be excluded